app/models/position_opening.rb

```diff
@@ -1,134 +1,224 @@
+          type: 'synonym',
+          synonyms: SYNONYMS
+        }
+      },
+      analyzer: {
+        custom_analyzer: {
+          type: 'custom',
+          tokenizer: 'whitespace',
+          filter: %w(standard lowercase synonym snowball)
+        }
+      }
+    }
+  }
 
-  index_name("#{Elasticsearch::INDEX_NAME}")
+  settings index: SETTINGS do
+    mappings dynamic: 'false' do
+      indexes :type, type: 'keyword'
+      indexes :source, type: 'keyword'
+      indexes :tags, type: 'text', analyzer: 'keyword'
+      indexes :external_id, type: 'integer', store: true
+      indexes :position_title, type: 'text', analyzer: 'custom_analyzer', term_vector: 'with_positions_offsets', store: true
+      indexes :organization_id, type: 'text', analyzer: 'keyword'
+      indexes :organization_name, type: 'keyword', index: false
 
-  MAX_RETURNED_DOCUMENTS = 100
-  SYNONYMS = ["information technology, it, tech, computer", "teacher, teaching", "certified nursing assistant, cna", "rn, registered nurse", "lpn, licensed practical nurse", "lvn, licensed vocational nurse", "pa, physician assistant", "custodial, janitor, custodian", "cys, child youth services", "clerk, clerical", "physician, doctor", "linguist, language", "tv, television", "legal, attorney", "counselor, counseling, therapy, therapist", "green, environment, environmental", "forester, forestry", "technical, technician", "technology, technologist", "electronic, electrical", "architect, architecture", "cypa, child and youth program assistant, childcare", "tso, transportation security officer"].freeze
+      indexes :locations, type: 'nested' do
+        indexes :city, type: 'text', analyzer: 'simple'
+        indexes :state, type: 'text', analyzer: 'keyword'
+        indexes :geo, type: 'geo_point'
+      end
+
+      indexes :start_date, type: 'date', format: 'YYYY-MM-dd'
+      indexes :end_date, type: 'date', format: 'YYYY-MM-dd'
+      indexes :minimum, type: 'float'
+      indexes :maximum, type: 'float'
+      indexes :position_offering_type_code, type: 'integer'
+      indexes :position_schedule_type_code, type: 'integer'
+      indexes :rate_interval_code, type: 'text', analyzer: 'keyword'
+      indexes :id, type: 'keyword', index: false
+      indexes :timestamp, type: 'date'
+      indexes :ttl, type: 'date'
+    end
+  end
 
   class << self
 
+    def client
+      @client ||= PositionOpening.__elasticsearch__.client
+    end
+
     def create_search_index
-      Tire.index index_name do
-        create(
-          settings: {
-            index: {
-              analysis: {
-                analyzer: { custom_analyzer: { type: 'custom', tokenizer: 'whitespace', filter: %w(standard lowercase synonym snowball) } },
-                filter: { synonym: { type: 'synonym', synonyms: SYNONYMS } }
-              }
-            }
-          },
-          mappings: {
-            position_opening: {
-              _timestamp: { enabled: true },
-              _ttl: { enabled: true },
```

Comment:
I think there is something important going on here regarding the special `_ttl` field being changed to a "regular" field called `ttl`. The Elasticsearch cluster will no longer get rid of documents whose `ttl` duration has passed.

Were you aware of this issue @yih-en, and if so, did you have a plan to replace this functionality?

Reply:
We can set up a cron job to delete the expired/invalid job posts: [See here](https://github.com/GSA/jobs_api/pull/17/files#diff-681110cebeffad0ee72c994adfb4c087R293)